gammapy/datasets/fermi.py

```diff
@@ -216,6 +218,166 @@ def fetch_fermi_diffuse_background_model(filename='gll_iem_v02.fit'):
 
     return filename
 
+class Fermi3FGLObject(object):
+    """TODO: Doccomments
+    """
+
+    from astropy.units import Quantity
+    from gammapy.spectrum import energy_bin_centers_log_spacing
+
+    # Fermi catalog is lazily loaded on first access
+    # and cached at class level (not instance level)
+    fermi_cat = None
+
+    x_bins_edges = Quantity([30, 100, 300, 1000, 3000, 10000, 100000], 'MeV')
+
+    x_bins = Quantity(x_bins_edges, 'MeV')
+
+    x_cens = energy_bin_centers_log_spacing(x_bins)
+
+    y_labels = ['Flux30_100', 'Flux100_300', 'Flux300_1000',
+                'Flux1000_3000', 'Flux3000_10000', 'Flux10000_100000']
+
+    def __init__(self, source_name):
+        fermi_cat = self.get_fermi_cat()
+        self.name_3FGL = source_name
+        self.catalog_index = np.where(self.fermi_cat[1].data['Source_Name'] == source_name)[0][0]
+        self.cat_row = fermi_cat[1].data[self.catalog_index]
+        self.ra = self.cat_row['RAJ2000']
+        self.dec = self.cat_row['DEJ2000']
+        self.glon = self.cat_row['GLON']
+        self.glat = self.cat_row['GLAT']
+        self.int_flux = self.cat_row['Flux_Density']
+        self.unc_int_flux = self.cat_row['Unc_Flux_Density']
+        self.spec_type = self.cat_row['SpectrumType']
+        self.pivot_en = self.cat_row['PIVOT_ENERGY']
+        self.spec_index = self.cat_row['Spectral_Index']
+        self.unc_spec_index = self.cat_row['Unc_Spectral_Index']
+        self.beta = self.cat_row['beta']
+        self.unc_beta = self.cat_row['unc_beta']
+        self.cutoff = self.cat_row['Cutoff']
+        self.unc_cutoff = self.cat_row['Unc_Cutoff']
+        self.exp_index = self.cat_row['Exp_Index']
+        self.unc_exp_index = self.cat_row['Unc_Exp_Index']
+        self.signif = self.cat_row['Signif_Avg']
+
+    @classmethod
+    def get_fermi_cat(cls):
+        if not cls.fermi_cat:
+            cls.fermi_cat = fetch_fermi_catalog('3FGL')
+        return cls.fermi_cat
+
+    def plot_spectrum(self, ax=None):
+        import matplotlib.pyplot as plt
+        from gammapy.extern.stats import gmean
+        from astropy.modeling.models import PowerLaw1D, LogParabola1D, ExponentialCutoffPowerLaw1D
+
+        ax = plt.gca() if ax is None else ax
+
+        valid_indices = []
+        flux = np.array([self.cat_row[self.y_labels[i]] for i in (range(0,np.size(self.y_labels)))])
+        flux_lower_bound = np.array([self.cat_row["Unc_" + self.y_labels[i]][0] for i in (range(1,np.size(self.y_labels)))])
+        arr1 = np.where(np.isnan(flux) == False)[0]
+        arr2 =np.where(np.isnan(flux_lower_bound) == False)[0]
+        raw_input(arr2)
```

Comment:
Is this how you debug?
I've never used `raw_input` and it doesn't even exist on Python 3.

Just in case it's useful to you: I debug by putting `import IPython; IPython.embed()` on the line before where I get an error or where I want to continue coding. This drops you into an interactive IPython session right at this point, and you can look around and do some coding. And when it works, call `%history` and copy & paste the relevant code into your editor.